### ISSUE TITLE
Menu over Content

### ISSUE THREAD

**Issue by tehsphinx** [fyne-io/fyne#1973]
### Describe the bug:
When using `Resize` to set a reasonable size for the window before showing it, the menu overlaps the content.

### To Reproduce:
Run the sample code below using `go run -tags=no_native_menus .` (tested on Mac)

### Screenshots:
![image](https://user-images.githubusercontent.com/6624571/108380056-caa34500-7206-11eb-90d7-e8af8cc740a1.png)

### Example code:
```go
package main

import (
	"fyne.io/fyne/v2"
	"fyne.io/fyne/v2/app"
	"fyne.io/fyne/v2/widget"
)

func main() {
	a := app.New()

	w := a.NewWindow("Test App")
	w.Resize(fyne.Size{
		Width:  400,
		Height: 300,
	})
	w.SetMainMenu(fyne.NewMainMenu(fyne.NewMenu("Menu")))
	w.SetContent(widget.NewLabel("Label"))

	w.ShowAndRun()
}
```

### Device:
 - MacOS (but should be the same on all systems when using non-native menu)
 - Big Sur 11.2.1
 - go version go1.16 darwin/amd64
 - fyne.io/fyne/v2 v2.0.0

**Comment by tehsphinx** [comment]
It works if the `Resize` Command is moved below setting the content and main menu. Should probably still be fixed.

**Comment by andydotxyz** [comment]
This should have been fixed in #1138 - it felt like the same as issue #1105.
Looks like it has reappeared :(

**Comment by andydotxyz** [comment]
On `develop` for testing :)
